monai/losses/aucm_loss.py

```diff
@@ -0,0 +1,173 @@
+
+
+class AUCMLoss(_Loss):
+    """
+    AUCM Loss with squared-hinge function: a novel loss function to directly optimize AUROC
+
+    The original paper: Large-scale Robust Deep AUC Maximization: A
+    New Surrogate Loss and Empirical Studies on Medical Image Classification
+    https://arxiv.org/abs/2012.03173
+    """
+
+    def __init__(
+        self,
+        margin: float = 1.0,
+        pos_weight: Optional[Tensor] = None,
+        num_classes: int = 2,
+        to_onehot_y: bool = False,
+        sigmoid: bool = False,
+        softmax: bool = False,
+        other_act: Optional[Callable] = None,
+        reduction: Union[LossReduction, str] = LossReduction.MEAN,
+    ):
+        """
+        Args:
+            margin: margin term for AUCM loss, default 1.0, range [0, 1]
+            imratio: imbalance ratio, ratio of positive and negative samples, default None
+            num_classes: number of classes, default 2
+            to_onehot_y: whether to convert `y` into the one-hot format. Defaults to False.
+            sigmoid: if True, apply a sigmoid function to the prediction.
+            softmax: if True, apply a softmax function to the prediction.
+            other_act: if don't want to use `sigmoid` or `softmax`, use other callable function to execute
+                other activation layers, Defaults to ``None``. for example:
+                `other_act = torch.tanh`.
+            reduction: {``"none"``, ``"mean"``, ``"sum"``}
+                Specifies the reduction to apply to the output. Defaults to ``"mean"``.
+
+                - ``"none"``: no reduction will be applied.
+                - ``"mean"``: the sum of the output will be divided by the number of elements in the output.
+                - ``"sum"``: the output will be summed.
+            gpu: if True, use GPU to calculate. Defaults to False.
+
+
+        Raises:
+            ValueError: margin must be in [0, 1]
+            TypeError: When ``other_act`` is not an ``Optional[Callable]``.
+            ValueError: and When more than 1 of [``sigmoid=True``, ``softmax=True``, ``other_act is not None``].
+                Incompatible values.
+        """
+        super().__init__(reduction=LossReduction(reduction).value)
+
+        if margin < 0 or margin > 1:
+            raise ValueError("imratio must be between 0 and 1")
+        if other_act is not None and not callable(other_act):
+            raise TypeError(f"other_act must be None or callable but is {type(other_act).__name__}.")
+        if int(sigmoid) + int(softmax) + int(other_act is not None) > 1:
+            raise ValueError("Incompatible values: more than 1 of [sigmoid=True, softmax=True, other_act is not None].")
+
+        self.margin = margin
+
+        self.p = pos_weight
+        self.num_classes = num_classes
+        if self.p:
+            assert len(self.p) == self.num_classes, "imratio must be a list of length num_classes"
+        else:
+            self.p = [0.0] * self.num_classes
```

Comment:
Can you fix the type of `[0.0]` due to the type flake8 error? Thanks!
https://github.com/Project-MONAI/MONAI/runs/8106129023?check_suite_focus=true